exposed-core/src/main/kotlin/org/jetbrains/exposed/dao/id/IdTable.kt

```diff
@@ -37,19 +37,49 @@ abstract class IdTable<T : Comparable<T>>(name: String = "") : Table(name) {
 
     /** All base columns that make up this [IdTable]'s identifier column. */
     val idColumns = HashSet<Column<out Comparable<*>>>()
+        get() = field.validate()
+
+    private var idColumnsValidated: Boolean = false
+    private var inTableInitialization: Boolean = false
+
+    private fun <F : HashSet<Column<out Comparable<*>>>> F.validate(): F {
+        if (idColumnsValidated || inTableInitialization) return this
+
+        val primaryKeyColumns = primaryKey?.columns?.toSet().orEmpty()
+        return when {
+//            primaryKeyColumns.isEmpty() || minus(primaryKeyColumns).isNotEmpty() -> {
+//                val message = "Table definition must include a primary key with all id columns: ${joinToString { it.name }}"
+//                exposedLogger.error(message)
+//                error(message)
+//            }
```

Comment:
Validation is dependent on timing in table definition (whether `entityId()` gets invoked first, or maybe primary key, so I opted to check on every access and prevent validation only if it's been confirmed.

I thought the above logic made sense, that a table with an `entityId()` column must have it as a primary key, but there are a couple examples in our tests that do not follow this rule and do not have a primary key at all. I will remove comments if we agree to merge this PR.